internal/server/auth_handlers.go

```diff
@@ -136,6 +173,34 @@ func (h *auth) Login(c echo.Context) error {
 	return c.JSON(http.StatusOK, login)
 }
 
+func (h *auth) LoginPKCE(c echo.Context) error {
+	// Filter params
+	var params service.LoginParams
+	if err := c.Bind(&params); err != nil {
+		log.Println("Could not get parameters:", err)
+		return c.JSON(http.StatusBadRequest, sferror.New("Could not get credentials."))
+	}
+	params.UserAgent = c.Request().UserAgent()
+	params.Session = currentSession(c)
+
+	if params.Email == "" || params.Password == "" || params.CodeVerifier == "" {
+		return c.JSON(http.StatusUnauthorized, sferror.New("Invalid login credentials."))
+	}
+
+	hash := sha256.Sum256([]byte(params.CodeVerifier))
+	base64_hash := fmt.Sprintf("%x", hash[:])
```

Comment:
```suggestion
	hexHash := fmt.Sprintf("%x", hash)
```
By the way, is there a reason to encode the hash in hexa and then in base64?

Reply:
Hi ! because that's what the original client is doing : https://github.com/standardnotes/snjs/blob/ef2d0d9d46cc7c0251bb496a2afbd71cbb354ed8/packages/sncrypto-node/src/CryptoNode.ts#L58 (added with a call to base64URLEncode, later in the auth part)